### ISSUE TITLE
[Bug]: ValueError

### ISSUE THREAD

**Issue by miguelangel-nubla** [custom-components/ble_monitor#1199]
### What happened?

Lots of `ValueError` in the logs.
```
ValueError: Sensor sensor.ble_heart_rate_f5f5da96f7bc has device class 'None', state class 'measurement' unit 'bpm' and suggested precision '0' thus indicating it has a numeric value; however, it has the non-numeric value: 'unavailable' (<class 'str'>)
```

Already tried removing the integration, restart HA, install integration, restart HA.

### Relevant log output

```shell
2023-06-15 22:18:33.298 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform ble_monitor
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 583, in state
numerical_value = float(value) # type:ignore[arg-type]
^^^^^^^^^^^^
ValueError: could not convert string to float: 'unavailable'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 471, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 749, in _async_add_entity
await entity.add_to_platform_finish()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 850, in add_to_platform_finish
self.async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 590, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 654, in _async_write_ha_state
state = self._stringify_state(available)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 596, in _stringify_state
if (state := self.state) is None:
^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 585, in state
raise ValueError(
ValueError: Sensor sensor.ble_heart_rate_f5f5da96f7bc has device class 'None', state class 'measurement' unit 'bpm' and suggested precision '0' thus indicating it has a numeric value; however, it has the non-numeric value: 'unavailable' (<class 'str'>)
2023-06-15 22:18:33.311 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform ble_monitor
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 583, in state
numerical_value = float(value) # type:ignore[arg-type]
^^^^^^^^^^^^
ValueError: could not convert string to float: 'unavailable'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 471, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 749, in _async_add_entity
await entity.add_to_platform_finish()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 850, in add_to_platform_finish
self.async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 590, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 654, in _async_write_ha_state
state = self._stringify_state(available)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 596, in _stringify_state
if (state := self.state) is None:
^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 585, in state
raise ValueError(
ValueError: Sensor sensor.ble_steps_f5f5da96f7bc has device class 'None', state class 'total' unit 'steps' and suggested precision '0' thus indicating it has a numeric value; however, it has the non-numeric value: 'unavailable' (<class 'str'>)
2023-06-15 22:18:33.335 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 583, in state
numerical_value = float(value) # type:ignore[arg-type]
^^^^^^^^^^^^
ValueError: could not convert string to float: 'unavailable'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 970, in _async_registry_updated
self.async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 590, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 654, in _async_write_ha_state
state = self._stringify_state(available)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 596, in _stringify_state
if (state := self.state) is None:
^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 585, in state
raise ValueError(
ValueError: Sensor sensor.ble_heart_rate_f5f5da96f7bc has device class 'None', state class 'measurement' unit 'bpm' and suggested precision '0' thus indicating it has a numeric value; however, it has the non-numeric value: 'unavailable' (<class 'str'>)
2023-06-15 22:18:33.343 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 583, in state
numerical_value = float(value) # type:ignore[arg-type]
^^^^^^^^^^^^
ValueError: could not convert string to float: 'unavailable'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 471, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 749, in _async_add_entity
await entity.add_to_platform_finish()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 850, in add_to_platform_finish
self.async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 590, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 654, in _async_write_ha_state
state = self._stringify_state(available)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 596, in _stringify_state
if (state := self.state) is None:
^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 585, in state
raise ValueError(
ValueError: Sensor sensor.ble_heart_rate_f5f5da96f7bc has device class 'None', state class 'measurement' unit 'bpm' and suggested precision '0' thus indicating it has a numeric value; however, it has the non-numeric value: 'unavailable' (<class 'str'>)
2023-06-15 22:18:33.356 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 583, in state
numerical_value = float(value) # type:ignore[arg-type]
^^^^^^^^^^^^
ValueError: could not convert string to float: 'unavailable'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 970, in _async_registry_updated
self.async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 590, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 654, in _async_write_ha_state
state = self._stringify_state(available)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 596, in _stringify_state
if (state := self.state) is None:
^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 585, in state
raise ValueError(
ValueError: Sensor sensor.ble_steps_f5f5da96f7bc has device class 'None', state class 'total' unit 'steps' and suggested precision '0' thus indicating it has a numeric value; however, it has the non-numeric value: 'unavailable' (<class 'str'>)
2023-06-15 22:18:33.365 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 583, in state
numerical_value = float(value) # type:ignore[arg-type]
^^^^^^^^^^^^
ValueError: could not convert string to float: 'unavailable'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 471, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 749, in _async_add_entity
await entity.add_to_platform_finish()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 850, in add_to_platform_finish
self.async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 590, in async_write_ha_state
self._async_write_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 654, in _async_write_ha_state
state = self._stringify_state(available)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 596, in _stringify_state
if (state := self.state) is None:
^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 585, in state
raise ValueError(
ValueError: Sensor sensor.ble_steps_f5f5da96f7bc has device class 'None', state class 'total' unit 'steps' and suggested precision '0' thus indicating it has a numeric value; however, it has the non-numeric value: 'unavailable' (<class 'str'>)
```

**Comment by Ernst79** [comment]
This is an bug in Home Assistant, please add your info in this issue 

https://github.com/home-assistant/core/issues/86948

**Comment by sesame-foxes** [comment]
The problem still alive？

**Comment by Ernst79** [comment]
Can you try with 12.2.2-beta? I think it should be fixed with that version

**Comment by miguelangel-nubla** [comment]
With 12.2.2-beta:
```
2023-07-12 13:24:08.530 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_rssi_fda50693a4e24fb1afcfc6eb07647825 (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('signal_strength') it is using; expected one of ['dB', 'dBm']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-12 13:24:08.534 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_measured_power_fda50693a4e24fb1afcfc6eb07647825 (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('signal_strength') it is using; expected one of ['dB', 'dBm']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-12 13:24:19.922 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_illuminance_c47c8d6b29cd (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('illuminance') it is using; expected one of ['lx']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-12 13:24:19.924 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_rssi_c47c8d6b29cd (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('signal_strength') it is using; expected one of ['dB', 'dBm']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-12 13:24:21.289 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_moisture_c47c8d6b29cd (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('humidity') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
```

**Comment by Ernst79** [comment]
Thanks for testing. I can't reproduce your issue, but I build in an additional check to not restore the data if the native_value is None. I think that will fix the warnings you have. Can you check it again with 12.2.3-beta?

**Comment by bentolor** [comment]
Thank you, @Ernst79, for you valuable contribution to the HASS integration landscape.

I upgraded to latest HASS today and I'm getting the same flood of errors and non-working entities right now. I'm on ble_monitor 12.3.0 installed via HACS.

Based on my quick glance of the issues here, you'd assume it should be resolved already with  12.3.0? 

```
2023-07-14 21:19:59.026 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_humidity_a4c1384ab2b2 (<class 'custom_components.ble_monitor.sensor.HumiditySensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('humidity') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:19:59.031 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_battery_a4c1384ab2b2 (<class 'custom_components.ble_monitor.sensor.BatterySensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('battery') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:19:59.033 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_voltage_a4c1384ab2b2 (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('voltage') it is using; expected one of ['V', 'mV']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:19:59.415 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_humidity_outdoor (<class 'custom_components.ble_monitor.sensor.HumiditySensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('humidity') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:19:59.417 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_battery_outdoor (<class 'custom_components.ble_monitor.sensor.BatterySensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('battery') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:19:59.418 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_voltage_outdoor (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('voltage') it is using; expected one of ['V', 'mV']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:19:59.419 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_rssi_outdoor (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('signal_strength') it is using; expected one of ['dBm', 'dB']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:19:59.467 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_humidity_a4c138b58227 (<class 'custom_components.ble_monitor.sensor.HumiditySensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('humidity') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:19:59.469 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_battery_a4c138b58227 (<class 'custom_components.ble_monitor.sensor.BatterySensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('battery') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:19:59.470 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_voltage_a4c138b58227 (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('voltage') it is using; expected one of ['V', 'mV']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:19:59.548 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_humidity_a4c138fc0755 (<class 'custom_components.ble_monitor.sensor.HumiditySensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('humidity') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:19:59.549 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_battery_a4c138fc0755 (<class 'custom_components.ble_monitor.sensor.BatterySensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('battery') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:19:59.549 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_voltage_a4c138fc0755 (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('voltage') it is using; expected one of ['V', 'mV']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:20:01.057 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_humidity_a4c138bf7c97 (<class 'custom_components.ble_monitor.sensor.HumiditySensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('humidity') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:20:01.057 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_battery_a4c138bf7c97 (<class 'custom_components.ble_monitor.sensor.BatterySensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('battery') it is using; expected one of ['%']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:20:01.057 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_voltage_a4c138bf7c97 (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('voltage') it is using; expected one of ['V', 'mV']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-14 21:20:01.058 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_rssi_a4c138bf7c97 (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('signal_strength') it is using; expected one of ['dBm', 'dB']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

```

**Comment by Ernst79** [comment]
Yes, it should have been fixed, but apparently not. Strange thing is that I don't see these errors myself, while I have similar devices. I'll look into it a.s.a.p.

**Comment by bentolor** [comment]
@Ernst79 I have to revert the claim: Seems I do get values again. Obviously I didn't get some of them when I was still on 12.2.0. Nevertheless: The warnings on the missing unit of the entities.

My HASS is running for quite some time an ble_monitor together with Xiaomi devices was the very first sensors I started with. I tried to check the HASS database content with `sqlitebrowser`, but it appears to me everything is ok on that site.

Below i.e. one of the `state_attributes` as in db.

````
{
    "battery_level": 54,
    "device_class": "signal_strength",
    "firmware": "ATC (Custom)",
    "friendly_name": "Balkon RSSI",
    "last_mean_of": 2,
    "last_packet_id": 214,
    "mac_address": "A4:C1:38:3A:41:48",
    "mean": -87.5,
    "median": -87.5,
    "sensor_type": "ATC",
    "state_class": "measurement",
    "unit_of_measurement": "dBm",
    "uuid": "A4:C1:38:3A:41:48"
}
````

So the urgency might be lower than I initially expected.

Thanks. @Ernst79 for your kind and prompt reply!

**Comment by Ernst79** [comment]
Ok, so you only have the warnings, but the sensors do update normally. Two questions
- do you get these warnings regularly, or only after a restart
- can you try a second restart of HA to see if you still get these messages?

thanks

**Comment by bentolor** [comment]
Hi @Ernst79 : I've already restarted my instance multiple times as side-effect of my investigations, so that doesn't seem to work.

Regarding your first question: It seems I get the warning the first time the value is read. So they appear after restart but not immediately.

**Comment by Ernst79** [comment]
Hmmm, ok. Last question. Are the states restored immediately after a restart, or is it unavailable till the first message arrives? Btw, I assume you have turned on the option to restore the state in the BLE monitor options

**Comment by bentolor** [comment]
@Ernst79 No: Currently only checked option is the MAC/hci of the BT adapter. "Restore state after a restart" is not yet. I was not even aware of this UI screen as I started 100% via `configuration.yaml` and even manually declaring the sensors.

Also: States are unavailable after restart until first value is read.

**Comment by Elarion245** [comment]
I am also lacking the units of all devices since upgrading to 12.3.0 and 2023.7.2. I cannot say which change caused this.

**Comment by Elarion245** [comment]
For the time being, a workaround to manually re-add the missing units without lots of code is to add the following to configuration.yaml (adapt according to what your devices provide):
```
    sensor.ble_rssi_*:
      unit_of_measurement: "dBm"
    sensor.ble_illuminance_*:
      unit_of_measurement: "lx"
    sensor.ble_conductivity_*:
      unit_of_measurement: "µS/cm"
    sensor.ble_temperature_*:
      unit_of_measurement: "°C"
    sensor.ble_moisture_*:
      unit_of_measurement: "%"
```

**Comment by miguelangel-nubla** [comment]
@Ernst79

Deleted integration, updated to 12.3.0, restarted HA, added integration again:

![config](https://github.com/custom-components/ble_monitor/assets/866596/520235f0-4cba-4d10-b154-3e81150faa27)

![warnings](https://github.com/custom-components/ble_monitor/assets/866596/520020df-a31b-4c9f-88b4-9e98e59771ae)

**Comment by Ernst79** [comment]
I was finally able to reproduce the issue. I think I found the problem and a solution. Fix is on its way in #1208

**Comment by Ernst79** [comment]
Please try 12.3.1, this version should fix this issue.

**Comment by Elarion245** [comment]
Thanks for the effort! I've updated to 12.3.1 and removed the custom_glob workaround. Sadly things haven't changed. My MiFlora sensors are still lacking units and also the log messages are still there...

**Comment by Ernst79** [comment]
Did you restart Home Assistant after the update?

**Comment by Elarion245** [comment]
Yes I did. Installed update and restarted HA (not Host though)

**Comment by Ernst79** [comment]
Hmmm, very strange. On my end, error are gone, both with restore state option enabled and disabled. All units of measurements are present.

can you enable debugging for ble monitor and post a log here after a restart of ha till the data is updated?

**Comment by miguelangel-nubla** [comment]
I have fewer warnings now, and took 8+ hours to show up:
```
2023-07-17 08:39:48.443 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_rssi_910a0c6b9d104d129c9bee339b478327 (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('signal_strength') it is using; expected one of ['dB', 'dBm']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-17 08:39:48.450 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_measured_power_910a0c6b9d104d129c9bee339b478327 (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('signal_strength') it is using; expected one of ['dB', 'dBm']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
```

For now it only shows up with rssi and measured_power, which I guess are active scan only entities?

@Ernst79 sent the full log via email.

**Comment by Ernst79** [comment]
@miguelangel-nubla Thanks for the logs. I see that the errors in your log only happen during the restart of the scanning process. The HCI scanning is restarted every minute (default period). Somehow, RSSI measurements and measured power (both in dBm) fail during that period. Strangely, @xelemorf in #1209 is saying that he still has sensors without unit of measurements. I'm still a bit lost where things go wrong, as I don't see any of these errors. 

```
2023-07-17 07:21:13.510 DEBUG (Thread-47) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing.
2023-07-17 07:21:13.514 DEBUG (Thread-47) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2023-07-17 07:21:13.514 DEBUG (Thread-47) [custom_components.ble_monitor] 1163 HCI events processed for previous period
2023-07-17 07:21:13.514 DEBUG (Thread-47) [custom_components.ble_monitor] HCIdump thread: Run
2023-07-17 07:21:13.521 DEBUG (Thread-47) [custom_components.ble_monitor] HCIdump thread: BLEScanRequester._initialized is False for hci0,  waiting for connection...
2023-07-17 07:21:13.527 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_rssi_8063ae90d8b711e695980800200c9a66 (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('signal_strength') it is using; expected one of ['dB', 'dBm']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
2023-07-17 07:21:13.545 DEBUG (Thread-47) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2023-07-17 07:21:13.546 DEBUG (Thread-47) [custom_components.ble_monitor] HCIdump thread: BLEScanRequester._initialized is True for hci0,  connection established, send_scan_request succeeded.
2023-07-17 07:21:13.546 DEBUG (Thread-47) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2023-07-17 07:21:13.559 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.ble_measured_power_8063ae90d8b711e695980800200c9a66 (<class 'custom_components.ble_monitor.sensor.MeasuringSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('signal_strength') it is using; expected one of ['dB', 'dBm']; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
```

**Comment by pamatika** [comment]
I can confirm. Issue still persist. RUUVI tags temperature units is working but pressure, humidity, battery etc are not working.

**Comment by Elarion245** [comment]
Hello,

Please find below the debug log:
[Debug_Log.txt](https://github.com/custom-components/ble_monitor/files/12072447/Debug_Log.txt)

The interesting thing is that for sensors of the same type it rarely works for some parameters, but not on other sensors. The temperature always seems to come up with units:
![grafik](https://github.com/custom-components/ble_monitor/assets/7480315/74bae79e-a133-4d3a-8fa5-b21fd79ebc11)

**Comment by bentolor** [comment]
Thanks for your efforts to resolve this issue, @Ernst79 ! 
I just came back home and gave an upgrade & restart a try. 

The behaviour clearly changed: I don't have "Restore state after a restart" enabled. Now after restart, the first values become visible without any warnings. That's new.

Similar to @miguelangel-nubla expierence, some warnings show up _en block_ after 2,5h. Could it be due to persistent data? Really strange.

Nota bene: My setup contains a bluetooth proxy and I'm keeping longterm statistics virtually indefinitely as I want to be able to proof my humidity in my apartment.

**Comment by Elarion245** [comment]
I can confirm that the instructions provided by @torrvick indeed also fixed it for me! 
Thank you @torrvick ! How did you manage to find this out? I have also seen the > 100%, but would have never thought that this may be the reason. 

A minor modification I had to do was to replace the "sensor.ble_humidity%" part with "sensor.ble_mois%" as for the Xiaomi MiFlora sensors the percentage value is called moisture. Also, as I was a bit scared to just paste the DELETE, I replaced it with SELECT * first and then when I saw the records popping up, I replaced the SELECT * with DELETE which then really removed the records.
I will keep the SQLite Web Add-On installed... 

Thanks again!!

**Comment by torrvick** [comment]
@Elarion245, i'm glad it helped you) I don't know why, but 1 of 5 sensor (Xiaomi Mijia LYWSD03MMC) suddenly began to show crazy values in graph history between 5000 and 6000. Else 3 sensors showed normal values (but was without a percents too). 1 sensor meanwile showed percents (i'm not sure what was in database for this sensor, because 1 tried trick with removing history on one, it helped, then i did it for everyone at once). All sensors have the same firmware and the same settings. What about values, it seems like correct humidity multiplied by 100, because it really was about 50-60%

**Comment by Ernst79** [comment]
In #1210 there was still an issue with the restore state function, so I created another fix in 12.3.2-beta. I hope that this version will fix the remaining issues.
